Week1/homework/app.js

```diff
@@ -1,11 +1,170 @@
+      published: 2018,
+    },
+    war_with_grandpa: {
+      title: 'The War with Grandpa',
+      language: 'English',
+      author: 'Robert Kimmel Smith',
+      published: 2018,
+    },
+    ready_player_one: {
+      title: 'Ready Player One',
+      language: 'English',
+      author: 'Ernest Cline',
+      published: 2018,
+    },
+    darkest_minds: {
+      title: 'The Darkest Minds',
+      language: 'English',
+      author: 'Alexandra Bracken',
+      published: 2018,
+    },
+    boy_erased: {
+      title: 'Boy Erased',
+      language: 'English',
+      author: 'Garrard Conley',
+      published: 2018,
+    },
+    first_man: {
+      title: 'First Man',
+      language: 'English',
+      author: 'James R. Hansen',
+      published: 2016,
+    },
+    little_women: {
+      title: 'Little Women',
+      language: 'English',
+      author: 'Louisa May Alcott',
+      published: 2018,
+    },
+    ophelia: {
+      title: 'Ophelia',
+      language: 'English',
+      author: 'Lisa Klein',
+      published: 2018,
+    },
+  };
+
+  /* cSpell:enable */
+
+  // 1.7 Find and download book covers for each book and construct a new object which has as keys the book IDs again,
+  //  and as value the path to the image source.
+  const bookImages = {
+    maze_runner_death_cure: 'images/maze_runner_death_cure.jpg',
+    fifty_shades_freed: 'images/fifty_shades_freed.jpg',
+    annihilation: 'images/annihilation.jpg',
+    war_with_grandpa: 'images/war_with_grandpa.jpg',
+    ready_player_one: 'images/ready_player_one.jpg',
+    darkest_minds: 'images/darkest_minds.jpg',
+    boy_erased: 'images/boy_erased.jpg',
+    first_man: 'images/first_man.jpg',
+    little_women: 'images/little_women.jpg',
+    ophelia: 'images/ophelia.jpg',
+  };
+
+  // 1.6 Beautify your html page with css (use the style.css file for that), add sources and alts to each of the images.
+
+  // 1.5 - 1.8
+  function bookDetails(booksList) {
+    for (let i = 0; i < bookTitles.length; i++) {
+      const listItem = document.createElement('li');
+      const bookInfo = Object.keys(bookInformation);
+      // 1.8 giving each li item an id tag
+      listItem.setAttribute('id', bookTitles[i]);
+      listItem.className = 'list-item';
+      booksList.appendChild(listItem);
+      const titles = document.createElement('h3');
+      titles.innerText = bookInformation[bookInfo[i]].title;
+      listItem.appendChild(titles);
+      const authors = document.createElement('p');
+      authors.innerText = bookInformation[bookInfo[i]].author;
+      listItem.appendChild(authors);
+      const languages = document.createElement('p');
+      languages.innerText = bookInformation[bookInfo[i]].language;
+      listItem.appendChild(languages);
+      const publishing = document.createElement('p');
+      publishing.innerText = bookInformation[bookInfo[i]].published;
+      listItem.appendChild(publishing);
+      const images = document.createElement('img');
```

Comment:
According to the spec, the book covers should only be added after the whole list is built, but finding the elements and adding the book cover dynamically.  (Yes it is a silly restriction, but it is more about the knowledge needed to make it happen than anything else 😉)

Reply:
I'll improve this one also.